lrpar/src/lib/ctbuilder.rs

```diff
@@ -146,6 +151,32 @@ impl Visibility {
     }
 }
 
+pub trait GrammarErrorHandler<LexerTypesT>
+where
+    LexerTypesT: LexerTypes,
+    usize: num_traits::AsPrimitive<LexerTypesT::StorageT>,
+{
+    fn grammar_src(&mut self, src: &str);
+    fn grammar_path(&mut self, path: &Path);
+    fn warnings_are_errors(&mut self, flag: bool);
```

Comment:
One thing missing here is that warnings are no longer printed, but there is no mechanism to print them from the trait.
there is also the `show_warnings` flag of `CTParserBuilder` we need to add if traits should print warnings.

One thing to consider is that we could change the `Ok(())` value of `results()` to return formatted warnings,
`CTParserBuilder` then just prints them out to `stderr`, or perhaps some `emit_warnings()` call?

Reply:
> One thing missing here is that warnings are no longer printed, but there is no mechanism to print them from the trait.

The trait doesn't mind if you print or don't :) I have a couple of possible ideas here:

1. The default implementation of the trait provided by the `CTBuilder`s is that they print warnings and/or panic pretty much as they do now.
2. We can provide an implementation of the trait which behaves as it does now but *before* it prints/panics also calls a user implementation of the trait.

Of these, 1) is the easiest to implement and to get right, so I'm inclined to start there, assuming we think it makes sense!